src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+            Using TestBase As New FileIOTests
+                Dim SourceFileNameWithPath As String = CreateTestFile(TestBase, SourceData)
+                Dim DestinationFileNameWithPath As String = IO.Path.Combine(TestBase.TestDirectory, "Select_Skip_this_file")
+                FileSystem.MoveFile(SourceFileNameWithPath, DestinationFileNameWithPath, showUI:=UIOption.AllDialogs, onUserCancel:=UICancelOption.DoNothing)
+                Assert.False(IO.File.Exists(SourceFileNameWithPath))
+                Assert.True(IO.File.Exists(DestinationFileNameWithPath))
+                Assert.True(HasExpectedData(DestinationFileNameWithPath, SourceData))
+                SourceFileNameWithPath = DestinationFileNameWithPath
+                DestinationFileNameWithPath = CreateTestFile(TestBase, DestData)
+                FileSystem.MoveFile(SourceFileNameWithPath, DestinationFileNameWithPath, showUI:=UIOption.AllDialogs, onUserCancel:=UICancelOption.ThrowException)
+                ' Make sure we did not override existing file
+                Assert.True(HasExpectedData(DestinationFileNameWithPath, DestData))
+                Assert.True(IO.File.Exists(SourceFileNameWithPath))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub RenameDirectory_Directory_NewName()
+            Using TestBase As New FileIOTests
+                ''' <exception cref="IO.FileNotFoundException">If directory does not point to an existing directory.</exception>
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Sub() FileSystem.RenameDirectory(IO.Path.Combine(TestBase.TestDirectory, "DoesNotExistDirectory"), "NewDirectory"))
+                Dim OrigDirectoryWithPath As String = IO.Path.Combine(TestBase.TestDirectory, "OriginalDirectory")
+                IO.Directory.CreateDirectory(OrigDirectoryWithPath)
+                ''' <exception cref="System.ArgumentException">If newName is Nothing or Empty String.</exception>
+                Assert.Throws(Of ArgumentNullException)(Sub() FileSystem.RenameDirectory(OrigDirectoryWithPath, ""))
+                Dim DirectoryNameWithPath As String = IO.Path.Combine(TestBase.TestDirectory, "DoesNotExist")
+                ''' <exception cref="System.ArgumentException">If contains path information.</exception>
+                Assert.Throws(Of ArgumentException)(Sub() FileSystem.RenameDirectory(OrigDirectoryWithPath, DirectoryNameWithPath))
+                FileSystem.RenameDirectory(OrigDirectoryWithPath, "NewFDirectory")
+                Dim NewFDirectoryPath As String = IO.Path.Combine(TestBase.TestDirectory, "NewFDirectory")
+                Assert.True(IO.Directory.Exists(NewFDirectoryPath))
+                Assert.False(IO.Directory.Exists(OrigDirectoryWithPath))
+                ''' <exception cref="IO.IOException">If directory points to a root directory or if there's an existing directory or an existing file with the same name.</exception>
```

Comment:
Consider removing `If directory points to a root directory or` since it looks like we're not testing that case.

Reply:
I can remove or should I try to add test, this may have to be OS specific.